sshuttle/client.py

```diff
@@ -643,9 +651,23 @@ def matches_acl(dstip, dstport, store_to_check):
 
     return False
 
+def ipInNetwork(ip, cidrBlock):
+    try:
+        network = ipaddress.ip_network(cidrBlock, False)
+        reqIp = ipaddress.ip_address(ip)
+        if reqIp in network:
+            return True
+    except:
+        log('Failed to parse CIDR block %s' % cidrBlock)
+    return False
+
 def tcp_connection_is_allowed_conditional(dstip, dstport, srcip, check_acl, check_sources):
 
     if check_sources:
+        # allow any IP from the two Microservice subnets
+        if ipInNetwork(srcip, MS1_CIDR) or ipInNetwork(srcip, MS2_CIDR):
```

Comment:
How did you test this condition?  How did you confirm that it didn't just use the if _excluded_sources statement below?

Reply:
So in order to test this, I tried to connect something running on the gateway side from the microservices box via gateway private address. As suggested by Zach, I stat a http sever in python on my gateway machine, which serve static files from the current directory and I tried to access those files(curl command) via gateway private address from microservices.

Reply:
but Luc's point is valid, how do you know you weren't hitting the ACL excluded block? That test would come from repeating exactly your test scenario but from one of the DC instances since they aren't in ACL excluded. 

I also believe you unit tested this function although it's still worth doing the full scenario test.

Reply:
@lucbertin @pythianrenaud : I did ran full scenario test as well. Try to connect from DCs to gateway via private address. I did get response and the logs are related to the new function. Didn't see logs from _excluded_sources. So that means it  didn't run _excluded_sources statement.